Patch-Source: https://code.qt.io/cgit/qbs/qbs.git/patch/?id=ee78fcb399ec5968a43dc0160ee40cdd5357cc1a
---
From ee78fcb399ec5968a43dc0160ee40cdd5357cc1a Mon Sep 17 00:00:00 2001
From: Ivan Komissarov <abbapoh@gmail.com>
Date: Sat, 1 Nov 2025 19:09:18 +0300
Subject: cmake: fix build with Qt 6.10

See QTBUG-87776, since Qt 6.9, QtCorePrivate was moved into
a separate package. Since 6.10, compatibility was also removed
and configuration fails if QtCorePrivate is not found using
find_package.

Change-Id: Ida62060606e2f92b6c38581c90b91951d8f1f310
Reviewed-by: Christian Kandeler <christian.kandeler@qt.io>
---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a04a0135d..5f1733b7a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,6 +50,9 @@ if (Qt6_FOUND)
         endif()
         add_library(Qt6Core5Compat ALIAS Qt6::Core5Compat)
     endif()
+    if (Qt6Core_VERSION VERSION_GREATER_EQUAL 6.9.0)
+        find_package(Qt${QT_VERSION_MAJOR} COMPONENTS CorePrivate REQUIRED)
+    endif()
 else()
   if(NOT TARGET Qt6Core5Compat)
     add_library(Qt6Core5Compat INTERFACE)
-- 
cgit v1.2.3
